pkg/restapi/operation/common/operation.go

```diff
@@ -0,0 +1,173 @@
+)
+
+// provider contains dependencies for the common controller operations
+// and is typically created by using aries.Context()
+type provider interface {
+	VDRIRegistry() vdriapi.Registry
+}
+
+// Operation contains basic common operations provided by controller REST API
+type Operation struct {
+	ctx      provider
+	handlers []operation.Handler
+}
+
+// New returns new common operations rest client instance
+func New(ctx provider) *Operation {
+	o := &Operation{ctx: ctx}
+	defer o.registerHandler()
+
+	return o
+}
+
+//
+
+// GetRESTHandlers get all controller API handler available for this service
+func (o *Operation) GetRESTHandlers() []operation.Handler {
+	return o.handlers
+}
+
+// registerHandler register handlers to be exposed from this protocol service as REST API endpoints
+func (o *Operation) registerHandler() {
+	// Add more protocol endpoints here to expose them as controller API endpoints
+	o.handlers = []operation.Handler{
+		support.NewHTTPHandler(createPublicDIDPath, http.MethodPost, o.CreatePublicDID),
+	}
+}
+
+// CreatePublicDID swagger:route POST /general/create-public-did general createPublicDID
+//
+// Creates a new Public DID....
+//
+// Responses:
+//    default: genericError
+//        200: createPublicDIDResponse
+func (o *Operation) CreatePublicDID(rw http.ResponseWriter, req *http.Request) {
+	var request CreatePublicDIDRequest
+
+	err := getQueryParams(&request, req.URL.Query())
+	if err != nil {
+		errors.SendHTTPBadRequest(rw, InvalidRequestErrorCode, err)
+		return
+	}
+
+	if request.CreatePublicDIDParams == nil || request.Method == "" {
+		errors.SendHTTPBadRequest(rw, InvalidRequestErrorCode, fmt.Errorf("invalid method name"))
+		return
+	}
+
+	logger.Debugf("creating public DID for method[%s]", request.Method)
+
+	doc, err := o.ctx.VDRIRegistry().Create(strings.ToLower(request.Method),
+		vdriapi.WithRequestBuilder(buildSideTreeRequest))
+	if err != nil {
+		errors.SendHTTPInternalServerError(rw, CreatePublicDIDError, err)
+		return
+	}
+
+	o.writeResponse(rw, CreatePublicDIDResponse{DID: doc})
+}
+
+// writeResponse writes interface value to response
+func (o *Operation) writeResponse(rw io.Writer, v interface{}) {
+	err := json.NewEncoder(rw).Encode(v)
+	// as of now, just log errors for writing response
+	if err != nil {
+		logger.Errorf("Unable to send error response, %s", err)
+	}
+}
+
+// getQueryParams converts query strings to `map[string]string`
+// and unmarshals to the value pointed by v by following
+// `json.Unmarshal` rules.
+func getQueryParams(v interface{}, vals url.Values) error {
+	// normalize all query string key/values
+	args := make(map[string]string)
+
+	for k, v := range vals {
+		if len(v) > 0 {
+			args[k] = v[0]
+		}
+	}
+
+	b, err := json.Marshal(args)
+	if err != nil {
+		return err
+	}
+
+	return json.Unmarshal(b, v)
+}
+
+// buildSideTreeRequest request builder for sidetree public DID creation
+func buildSideTreeRequest(didDoc *did.Doc) (io.Reader, error) {
```

Comment:
aries code should not know about sidetree

Reply:
Removed, I just added feature to inject optional header to http binding request. Using that header to inject side tree specific minimal headers, and it is working. 
The current create request in REST API follows `{header:xxy, payload:xxz}` pattern. We can modify it anytime in future to add support to other public DIDs.